File: POLYENCAPINHEBSTRACTION/BankAccount.php
<?php


class BankAccount {
    private $balance;
    //Constructor 
    public function __construct($balance) {
        $this->balance = 0;
    }
    //Deposit Method
    public function deposit($amount) {
        if ($amount > 0) {
            $this->balance += $amount;
            echo "You have recieved $" . $amount;
        } else {
            echo "Invalid deposit amount";
        }
    }
    //Withdraw Method
    public function withdraw($amount) {
        if($amount > 0 && $amount <= $this->balance) {
            $this->balance -= $amount;
            echo "You have withrawn $$amount" . "<br>";
        } else {
            echo "Insufficient funds / Invalid withdrawal amount";
        }
    }
    //Get balance
    public function getBalance() {
        return $this->balance;
    }
}

$b1 = new BankAccount(0);

$b1->deposit(500);

$b1->withdraw(200);

echo "Balance: ".$b1->getBalance();

File: OOPstudy/BankAccount.java
package OOPstudy;

public class BankAccount {
    private String accNumber;
    private String accHoldName;
    private double balance;
    
    public BankAccount(String accNumber, String accHoldName, double balance) {
        this.accNumber = accNumber;
        this.accHoldName = accHoldName;
        this.balance = balance;
    }

    public void deposit(double amount) {
        if (amount >= 0) {
            balance += amount;
            System.out.println("You have deposited $" + amount + " into your account! ");
            System.out.println("You have $" + balance + " in your wallet !");
        } else {
            System.out.println("Invalid deposit amount !");
            System.out.println("You cannot deposit $" + amount + " into your account");
        }
    }
    public void withdraw(double amount) {
        if (amount >= 0 && amount <= balance) {
            balance -= amount;
            System.out.println("You have succesfully withdrawn $" + amount + "into your wallet!");
            System.out.println("Your current balnce is $" + balance);
        } else {
            System.out.println("Invalid withdrawal amount !");
            System.out.println("You cannot withdraw $" + amount + " from your wallet $" + balance);
        }
    }

    public double getBalance() {
        return balance;
    }
    public String getAccNumber(){
        return accNumber;
    }
}
